### ISSUE TITLE
XGBoost MultiClass classification taking too long

### ISSUE THREAD

**Issue by woodrujm** [dmlc/xgboost#4079]
I'm trying to implement XGBoost multiclass for ~750 classes. The model has been training for hours (overnight) while the SKLearn RandomForest Implementation takes minutes.

**Comment by hcho3** [comment]
Gradient boosting has an inherent weakness in multi-class classification, when the number of class is high. The reason is that gradient boosting requires that you train `[number of iterations] * [number of classes]` trees, whereas random forest only requires `[number of iterations]` trees.

Try running the experiment with SKLearn gradient boosting and it would be slower than RandomForest.
